rust/src/checkpoints.rs

```diff
@@ -400,7 +400,7 @@ fn parquet_bytes_from_state(state: &DeltaTableState) -> Result<Vec<u8>, Checkpoi
 
     debug!("Writing to checkpoint parquet buffer...");
     // Write the Checkpoint parquet file.
-    let writeable_cursor = InMemoryWriteableCursor::default();
+    let writeable_cursor = Cursor::new(vec![]);
     let mut writer = ArrowWriter::try_new(writeable_cursor.clone(), arrow_schema.clone(), None)?;
```

Comment:
```suggestion
    let mut writer = ArrowWriter::try_new(&mut writeable_cursor, arrow_schema.clone(), None)?;
```

This bit fixes the checkpoint tests. The old `InMemoryWriteableCursor` had a buffer of type [Arc<Mutex<Cursor<Vec<u8>>>>](https://github.com/apache/arrow-rs/blob/33e298444f251258dd289c8377c68a80925ab0b4/parquet/src/util/cursor.rs#L139) which [it cloned](https://github.com/apache/arrow-rs/blob/33e298444f251258dd289c8377c68a80925ab0b4/parquet/src/util/cursor.rs#L171-L177) internally.